nipype/interfaces/cat12/preprocess.py

```diff
@@ -593,6 +593,142 @@ def _list_outputs(self):
+                            usedefault=True,
+                            field='addnoise',
+                            desc='Strength of additional noise in noise-free regions. Add  minimal  amount  of noise in regions without any noise to avoid image segmentation problems. This parameter defines the strength of additional noise as percentage of the average signal intensity.')
+
+    rician = traits.Enum(
+        0,
+        1,
+        field='rician',
+        usedefault=True,
+        desc='''Rician noise
+        MRIs  can  have  Gaussian  or  Rician  distributed  noise with uniform or nonuniform variance across the image. If SNR is high enough
+        (>3)  noise  can  be  well  approximated by Gaussian noise in the foreground. However, for SENSE reconstruction or DTI data a Rician
+        distribution is expected. Please note that the Rician noise estimation is sensitive for large signals in the neighbourhood and can lead to
+        artefacts, e.g. cortex can be affected by very high values in the scalp or in blood vessels.''')
+
+    replaceNANandINF = traits.Enum(
+        1,
+        0,
+        field='replaceNANandINF',
+        usedefault=True,
+        desc='Replace NAN by 0, -INF by the minimum and INF by the maximum of the image.'
+    )
+
+    NCstr = traits.Enum(
+        '-Inf',
+        2,
+        4,
+        field='nlmfilter.optimized.NCstr',
+        usedefault=True,
+        desc='''Strength of Noise Corrections
+        Strength  of  the  (sub-resolution)  spatial  adaptive    non local means (SANLM) noise correction. Please note that the filter strength is
+        automatically  estimated.  Change this parameter only for specific conditions. The "light" option applies half of the filter strength of the
+        adaptive  "medium"  cases,  whereas  the  "strong"  option  uses  the  full  filter  strength,  force sub-resolution filtering and applies an
+        additional  iteration.  Sub-resolution  filtering  is  only  used  in  case  of  high image resolution below 0.8 mm or in case of the "strong"
+        option. light = 2, medium = -Inf, strong = 4'''
+    )
+
+
+class CAT12SANLMDenoisingOutputSpec(TraitedSpec):
+
+    out_file = File(desc='out file')
+
+
+class CAT12SANLMDenoising(SPMCommand):
+    """
+    Spatially adaptive non-local means (SANLM) denoising filter
+
+    This  function  applies  an spatial adaptive (sub-resolution) non-local means denoising filter
+    to  the  data.  This  filter  will  remove  noise  while  preserving  edges. The filter strength is
+    automatically estimated based on the standard deviation of the noise.
+
+    This   filter   is  internally  used  in  the  segmentation  procedure  anyway.  Thus,  it  is  not
+    necessary (and not recommended) to apply the filter before segmentation.
+
+
+    Examples
+    --------
+    >>> from nipype.interfaces import cat12
+    >>> c = cat12.CAT12SANLMDenoising()
+    >>> c.inputs.in_files='sub-test_FLAIR.nii'
```

Comment:
Need to use files in `nipype/testing/data/`. You could create an empty file of this name, or just use an existing file.

```suggestion
    >>> c.inputs.in_files='anatomical.nii'
```